arch/x86/kernel/cpu/common.c

```diff
@@ -990,20 +991,23 @@ static void init_speculation_control(struct cpuinfo_x86 *c)
 
 	if (cpu_has(c, X86_FEATURE_AMD_IBRS)) {
 		set_cpu_cap(c, X86_FEATURE_IBRS);
-		set_cpu_cap(c, X86_FEATURE_MSR_SPEC_CTRL);
+		if (!hypervisor_is_type(X86_HYPER_QNX))
+			set_cpu_cap(c, X86_FEATURE_MSR_SPEC_CTRL);
 	}
 
 	if (cpu_has(c, X86_FEATURE_AMD_IBPB))
 		set_cpu_cap(c, X86_FEATURE_IBPB);
 
 	if (cpu_has(c, X86_FEATURE_AMD_STIBP)) {
 		set_cpu_cap(c, X86_FEATURE_STIBP);
-		set_cpu_cap(c, X86_FEATURE_MSR_SPEC_CTRL);
+		if (!hypervisor_is_type(X86_HYPER_QNX))
+			set_cpu_cap(c, X86_FEATURE_MSR_SPEC_CTRL);
```

Comment:
Why we also remove it for AMD processor?

Reply:
Yes.